src/main/scala/com/lightbend/sbt/MultiReleaseJarPlugin.scala

```diff
@@ -0,0 +1,100 @@
+package com.lightbend.sbt
+
+import sbt._
+import sbt.Keys._
+import sbt.plugins.JvmPlugin
+import sbt.{ AutoPlugin, Def, PluginTrigger, Plugins }
+
+import scala.collection.immutable
+
+object MultiReleaseJarPlugin extends AutoPlugin {
+
+  object MultiReleaseJarKeys {
+    val MultiReleaseJar = config("MultiReleaseJar") extend Compile
+    
+    val java9Directory = settingKey[File]("Where the java9 sources are")
+    val scala9Directory = settingKey[File]("Where the scala9 sources are")
+    val compileJava9 = taskKey[Seq[File]]("Compile Java9 sources into META-INF")
+    val metaInfVersionsTargetDirectory = settingKey[File]("Where the java9 classes should be written to." +
+      "Usually this is: ./target/scala-2.12/classes/META-INF/versions/9")
+  }
+  import MultiReleaseJarKeys._
+  
+  val autoImport = MultiReleaseJarKeys 
+  
+  override def requires: Plugins = JvmPlugin
+  override def trigger: PluginTrigger = allRequirements
+
+  override def projectConfigurations = Seq(MultiReleaseJar)
+  
+  // ---------------------------------------
+  // This could be taken from`java.util.jar.Attributes.MULTI_RELEASE`, but we don't want to requirerunning on JDK9! 
+  val MULTI_RELEASE_KEY = "Multi-Release"
+  // ---------------------------------------
+  
+  override def projectSettings = Seq(
+    
+//    // hijack source generators to compile the java9 sources
+//    sourceGenerators in Compile += Def.task { compileJava9.value }.taskValue,
+    
+    packageOptions in (Compile, packageBin) +=
+      Package.ManifestAttributes("Multi-Release" -> "true"),
+    
+    // "9" source directories
+    java9Directory := sourceDirectory.value / "main" / "java9",
+    scala9Directory := sourceDirectory.value / "main" / "scala9",
+    unmanagedSourceDirectories in Compile ++= Seq(
+      java9Directory.value,
+      scala9Directory.value
+    ), 
+
+    // target - we kind of 'inject' our sources into the right spot:
+    metaInfVersionsTargetDirectory := crossTarget.value / "classes" / "META-INF" / "versions" / "9",
+    
+    crossTarget in MultiReleaseJar := metaInfVersionsTargetDirectory.value, 
+    
+    sourceDirectory in MultiReleaseJar := java9Directory.value,
+    target in MultiReleaseJar := metaInfVersionsTargetDirectory.value,
+    
+    
+    // implementations:
+    
+    compileJava9 := {
+      implicit val log = streams.value.log
+      
+      val j9Dir = java9Directory.value
+      val cacheDir = crossTarget.value / "java9-compile-cache"
+      log.info(s"j9Dir = ${j9Dir}")
+      val targetDir = metaInfVersionsTargetDirectory.value
+      log.info(s"targetDir = ${targetDir}")
+      val cp = (dependencyClasspath in Compile).value.map(_.data)
+      log.info(s"cp = ${cp}")
+
+      val MetaInfDir = metaInfVersionsTargetDirectory.value
+      
+      (compile in MultiReleaseJar).value
+      
+      IO.createDirectory(MetaInfDir)
+      IO.append(MetaInfDir / "mock-example.class", "IT WAS ME ALL ALONG!")
```

Comment:
That's a mock of course. that's effectively what I need to do. All files in "9" directories, need to be emitted into `META-INF/versions/9/the-classes`.